src/common.h

```diff
@@ -369,6 +369,24 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get all possible prefixes from an arbitrary YANG string by iteratively returning specific substrings.
+ *
+ * The function looks for possible prefix at the beginning of the @p str_begin and returns it.
+ * If there is some string at the @p str_begin and then follows the prefix, then the string is returned.
+ * If @p str_begin does not contain any prefix, then the entire @p str_begin is returned.
+ * The @p next can be used as the @p str_begin in the next iteration.
+ *
+ * @param[in] str_begin Begin of the input string.
+ * @param[in] str_end Length of the @p str_begin. If set to NULL then the @p str_begin must be NULL-terminated string.
+ * @param[out] is_prefix Type of substring found. Set to 1 for prefix, otherwise 0.
+ * @param[out] str_next Rest of the string after the found substring.
+ * If the @p is_prefix is set to 1 then the colon character is skipped.
+ * If no string remains, it is set to NULL.
+ * @return Number of bytes (length) of the found substring starting at @p str_begin.
+ */
+size_t ly_string_next(const char *str_begin, const char *str_end, int *is_prefix, const char **str_next);
```

Comment:
Not sure if @lePici is able to come up with something accurate enough, I think we should help. The idea is that this function can take any string in a YANG module (hence called YANG string) such as an extension name but even an instance-identifier default value and iteratively return for the caller all possible prefixes (substrings that could be interpreted as a prefix if the whole string conforms to some subset of XPath). The use-case is quite simple, I think, but accurately describing this function is not. Can you @rkrejci suggest some specific improvements based on what I wrote?

Now that I think about, maybe I was forcing this function to be more generic than is needed because for cases such as the extension instance name, this function is not (and will likely never be) used because of strict syntax restrictions. Meaning it will be used only for YANG data values (and YANG schema default values). So maybe `YANG string` can be switched for `YANG value` and the function renamed to `ly_value_next`.

Reply:
I will use the names `ly_value_prefix_next` and `YANG value` for now.